### PR TITLE
Fix libxc download location

### PR DESCRIPTION
**Pull request by andysim**
## Description
Modify the download location away from the (temporarily?) defunct tddft.org, back to gitlab.

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: external/upstream/libxc/CMakeLists.txt [hunk 1]
```diff
@@ -15,9 +15,9 @@ else()
         # Default: use a stable release tarball of libxc. To use the
         # development version of libxc, instead, comment the URL line,
         # and uncomment the GIT lines.
-        URL http://www.tddft.org/programs/octopus/down.php?file=libxc/4.3.4/libxc-4.3.4.tar.gz
-        #GIT_REPOSITORY https://gitlab.com/libxc/libxc.git
-        #GIT_TAG master
+        #URL http://www.tddft.org/programs/octopus/down.php?file=libxc/4.3.4/libxc-4.3.4.tar.gz
+        GIT_REPOSITORY https://gitlab.com/libxc/libxc.git
+        GIT_TAG 4.3.4
```
Comment (loriab) [inline external/upstream/libxc/CMakeLists.txt]: actually, keep using `URL` and keep the `GIT_`* ones commented. I think there's a comparable interface in GL to the GH URL lines for other externals. lgtm can't build from `GIT_`, only from `URL`.

Comment (andysim) [reply external/upstream/libxc/CMakeLists.txt]: Ah, didn't realize that.  Will change it now..

**Comment by susilehtola** [comment]
tddft.org had been hacked but it's been up for several hours again now

**Comment by andysim** [comment]
FYI, @susilehtola the site has been unreachable for me all afternoon and still is.  Just letting you know in case it's a misconfigured firewall or something similar.

**Pull request merged**